File: src/Services/Identity/Identity.Api/Models/Mentor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Identity.Api.Models {
    public class Mentor {

        public MentorType MentorType { get; set; }

        public int TeacherId { get; set; }
        public Teacher Teacher { get; set; }

        public int StudentId { get; set; }
        public Student Student { get; set; }

    }
}

File: src/Web/Spa/src/app/models/mentor.ts
import { Teacher } from './teacher';
import { Student } from './student';
import { Room } from './room';

export class TeacherMentor{
    mentorType:string;
    teacher:Teacher;
    room:Room;
}

export class StudentMentor{
    mentorType:string;
    student:Student;
}